machine/dni/dni_3048up/rootconf/sysroot-bin/onie-sysinfo

```diff
@@ -0,0 +1,271 @@
+    fi
+
+}
+
+[ -r $lib_dir/sysinfo-arch ]     && . $lib_dir/sysinfo-arch
+[ -r $lib_dir/sysinfo-platform ] && . $lib_dir/sysinfo-platform
+
+args="hsSevimrpcfdatP"
+
+usage()
+{
+    echo "$(basename $0) [-${args}]"
+    cat <<EOF
+Dump ONIE system information.
+
+COMMAND LINE OPTIONS
+
+	The default is to dump the ONIE platform (-p).
+
+	-h
+		Help.  Print this message.
+
+	-s
+		Serial Number
+
+	-P
+		Part Number
+
+	-e
+		Management Ethernet MAC address
+
+	-v
+		ONIE version string
+
+	-i
+		ONIE vendor ID.  Print the ONIE vendor's IANA enterprise number.
+
+	-m
+		ONIE machine string
+
+	-r
+		ONIE machine revision string
+
+	-p
+		ONIE platform string.  This is the default.
+
+	-c
+		ONIE CPU architecture
+
+	-f
+		ONIE configuration version
+
+	-d
+		ONIE build date
+
+	-t
+		ONIE partition type
+
+	-S
+		ONIE silicon switch vendor
+
+	-a
+		Dump all information.
+EOF
+}
+
+sn=no
+mac=no
+version=no
+vendor_id=no
+machine=no
+machine_rev=no
+platform=no
+pn=no
+arch=no
+config_version=no
+build_date=no
+partition_type=no
+switch_asic=no
+all=no
+
+[ $# -eq 0 ] && platform=yes
+
+while getopts "$args" a ; do
+    case $a in
+        h)
+            usage
+            exit 0
+            ;;
+        s)
+            sn=yes
+            ;;
+        S)
+            switch_asic=yes
+            ;;
+        e)
+            mac=yes
+            ;;
+        v)
+            version=yes
+            ;;
+        i)
+            vendor_id=yes
+            ;;
+        m)
+            machine=yes
+            ;;
+        r)
+            machine_rev=yes
+            ;;
+        p)
+            platform=yes
+            ;;
+        P)
+            pn=yes
+            ;;
+        c)
+            arch=yes
+            ;;
+        f)
+            config_version=yes
+            ;;
+        d)
+            build_date=yes
+            ;;
+        t)
+            partition_type=yes
+            ;;
+        a)
+            all=yes
+            ;;
+        *)
+            echo "Unknown argument: $a"
+            usage
+            exit 1
+    esac
+done
+
+count=1
+print_val()
+{
+    if [ $count -gt 1 ] ; then
+        echo -n " "
+    fi
+    if [ -n "$1" ] ; then
+        echo -n "$1"
+    else
+        echo -n "unknown"
+    fi
+    count=$(( $count + 1 ))
+}
+
+if [ "$sn" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_serial_num)
+    print_val $val
+fi
+
+if [ "$pn" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_part_num)
+    print_val $val
+fi
+
+if [ "$mac" = "yes" ] || [ "$all" = "yes" ] ; then
+    val=$(get_ethaddr)
+    print_val $val
+fi
+
+for v in version vendor_id machine machine_rev platform arch config_version partition_type build_date switch_asic ; do
+    eval dump='$'$v
+    if [ "$dump" = "yes" ] || [ "$all" = "yes" ] ; then
+        eval val='$onie'_$v
+        print_val $val
+        if [ "$version" = "yes" ] || [ "$v" = "version" ] ; then
+        	uboot_v="U-Boot-""$(cat /dev/mtd0 | grep '3048UP V' | awk '{print $13}')"
```

Comment:
This is an awkward way of getting the u-boot version into the output of "onie-sysinfo -v".  This will be a nightmare to maintain going forward.

What happens when the core ONIE project updates the onie-sysinfo command?  Who will make sure to update this copy of the onie-sysinfo command?

There must be a better way of doing this.

The version string should come from the `/etc/machine.conf` file and the `onie_version` variable.  See these comments for how the version string is constructed:

https://github.com/opencomputeproject/onie/blob/master/build-config/Makefile#L197-L239

From those instructions you can add an arbitrary string to the onie version by setting `VENDOR_VERSION` in the machine.make file.  In this case you can add the u-boot version to that variable.

That way you do not need to create a copy of `onie-sysinfo`, which will be very difficult to maintain.